lib/gameboard/board.rb

```diff
@@ -0,0 +1,495 @@
+    #     #=> "O"
+    #     #=> "X"
+    #     #=> "X"...
+    #
+    def each_value(&block)
+      return enum_for(__method__) if block.nil?
+      each{|cell| yield(cell.value)}
+    end
+
+    # Public: An enumerator that yields each cells coordinates.
+    # Example
+    #
+    #   board.each { |cell| p cell }
+    #     #=> [0, 0]
+    #     #=> [1, 0]
+    #     #=> [2, 0]
+    #     #=> [0, 1]
+    #     #=> [1, 1]...
+    #
+    def each_coordinate(&block)
+      return enum_for(__method__) if block.nil?
+      each { |cell| yield(cell.coord.position) }
+    end
+
+    # Public: Return the value of the next available point given a coordinate
+    #         and a slope.
+    #
+    # point  - The starting coordinate pair.
+    # slope  - An [x, y] slope (e.g. [1,1] means over one row, up one column).
+    #          It must be an integer.
+    # coords - An optional boolean to specify returning a coordinate pair
+    #          instead of a value.
+    #
+    # Example
+    #   given_board =   -------------
+    #                 0 | 1 | 2 | 3 |
+    #                   -------------
+    #                 1 | A | B | C |
+    #                   -------------
+    #                 2 | X | Y | Z |
+    #                   -------------
+    #                   - 0 - 1 - 2 -
+    #
+    #   delta([1,1], [-1, 1])
+    #     #=> 1 ([0,2] if coord argument given)
+    #
+    #   delta([0,0], [2, 1])
+    #     #=> C ([2,1] if coord argument given)
+    #
+    def delta(point, slope, coords = false)
+      delta =  point.zip(slope).map {|point| point.reduce(:+) }
+      piece = board.find { |cell| cell.coord.position == delta }
+      raise "Off Grid" unless !!piece
+      coords ? piece.coord.position : piece.value
+    end
+
+    # Public: Return every diagonal on the gameboard. It does not require a
+    #         square board to function.
+    #
+    # coords: An optional boolean to specify returning an Array of coordinates
+    #         instead of values.
+    #
+    # Example
+    #      -------------
+    #    0 | 1 | 2 | 3 |
+    #      -------------
+    #    1 | A | B | C |
+    #      -------------
+    #    2 | X | Y | Z |
+    #      -------------
+    #      - 0 - 1 - 2 -
+    #
+    #
+    #   diagonal
+    #     #=> [
+    #          ["X", "B", 3], [1, "B", "Z"],
+    #          ["A", 2], ["A", "Y"], [1], ["X"],
+    #          ["Y", "C"], [2, "C"], ["Z"], [3]
+    #         ]
+    #
+    #   diagonal(true)
+    #     #=> [
+    #          [[0, 0], [1, 1], [2, 2]],
+    #          [[0, 2], [1, 1], [2, 0]],
+    #          [[0, 1], [1, 2]],
+    #          [[0, 1], [1, 0]],
+    #          [[0, 2]],
+    #          [[0, 0]],
+    #          [[1, 0], [2, 1]],
+    #          [[1, 2], [2, 1]],
+    #          [[2, 0]],
+    #          [[2, 2]]
+    #         ]
+    #
+    def diagonal(coords = false)
+      diagonals = []
+
+      height.times do |i|
+        diagonals << get_diagonal([0, i], coords)
```

Comment:
Also, you're using a primitive to represent a coordinate. Perhaps it would be overkill to convert it here, maybe even actively detrimental to clarity. But earlier I said to be uniform in dealing with this concept, so I think I should at least point out my hypocrisy and / or call attention to this possible exception. What do you think?